building 'brass/c++' on Mac OS

Sorry for posting multiple issues, I managed to solved everything, but I also thought it could help others who would try to do the same. I am trying to build BRASS in OS X El Capitain 10.11.4.
The problem arose while trying to build 'brass' with `sudo ./setup.sh /installation_path`.
`make -C cansam` works after adding pstreams/pstreams.h (it still wants it to be in the same folder although it's not present in the archive that setup.h unzips)
`make -C c++` goes fine till `g++ -Wall -Wextra -g -I../cansam -O2   -c -o rearrgroup.o rearrgroup.cpp`, where default Mac compiler clang++ throws multiple exceptions of this type: 

> error: invalid operands to binary expression

After that I could only proceed by switching to non-Mac g++ compiler, and was still getting multiple errors of this type:

> Undefined symbols for architecture x86_64

Then I tried copying all the dependencies from _brass-group.cpp_ and _rearrgroup.cpp_ into _feature.h_ and _rearrgroup.h_, and after that the whole compilation of c++ part went fine.
So I think it may make sense to copy these dependencies into .h files from the very beginning, if installing this package on Mac.

--- a/c++/rearrgroup.cpp
+++ b/c++/rearrgroup.cpp
@@ -35,6 +35,7 @@
 
 #include "rearrgroup.h"
 
+#include <ostream>
 #include <cstdlib>
 
 #include "cansam/exception.h"

